core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -0,0 +1,41 @@
+package org.infinispan;
+
+import org.infinispan.commons.util.CloseableIterator;
+
+import java.util.Iterator;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+/**
+ *
+ * @param <R>
+ */
+public interface CacheStream<R> extends Stream<R> {
+   // This would disable sending requests to all other remote nodes compared to one at a time
+   // This can reduce memory pressure on the originator node at the cost of performance.  This value is ignored
+   // when terminal operators that aren't key knowledgable.
```

Comment:
The only key knowledgable terminal operators are forEach and iterator.  Every other operation doesn't track keys in a rehash environment and only track segments.  Yeah I have to write javadoc on *everything* still :)